core/osrd-geom/src/main/java/fr/sncf/osrd/geom/LineString.java

```diff
@@ -16,6 +17,50 @@ public final class LineString {
     /** A cumulative list of N-1 distances between coordinates */
     private final double[] cumulativeLengths;
 
+    private static final double EPSILON = 1E-8;
+
+    /** Check if 2 arrays of double are equal, allowing for epsilon variation and also allowing extra elements in list
+     * that would just differ from previous element by epsilon.
+     * Additional parameter to allow an extra first value that would be close to 0 */
+    private static boolean equalsMergeEpsilon(double[] left, double[] right, boolean isFirstEpsilonOk) {
+        int i = 0, j = 0;
+        while (true) {
+            if (Math.abs(left[i] - right[j]) < EPSILON) {
+                // default consume from both arrays if equal
+                if (i + 1 >= left.length && j + 1 >= right.length) return true;
+                if (i + 1 < left.length) i++;
+                if (j + 1 < right.length) j++;
+            } else if ((i > 0 && Math.abs(left[i - 1] - left[i]) < EPSILON)
+                    || (isFirstEpsilonOk && i == 0 && Math.abs(left[0]) < EPSILON)) {
+                // else, if current left value is equal to previous left value, consume left
+                // also check if the list should be considered as starting by a virtual 0
+                i++;
+            } else if ((j > 0 && Math.abs(right[j - 1] - right[j]) < EPSILON)
+                    || (isFirstEpsilonOk && j == 0 && Math.abs(right[0]) < EPSILON)) {
+                // same for right
+                j++;
+            } else return false;
+        }
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) return false;
+        if (obj.getClass() != this.getClass()) return false;
+
+        final LineString other = (LineString) obj;
+        if (!equalsMergeEpsilon(bufferX, other.bufferX, false)) return false;
+        if (!equalsMergeEpsilon(bufferY, other.bufferY, false)) return false;
+        if (!equalsMergeEpsilon(cumulativeLengths, other.cumulativeLengths, true)) return false;
```

Comment:
Generally I'd prefer if `.equals` overloads don't include epsilons, with another method for equality with tolerance. Though that's fine in a branch like that.

Reply:
Cleaned in https://github.com/osrd-project/osrd/pull/6670/commits/6940ff8e4d992bddc66e23e095528ecd1bc9d0b1
I guess it's nice to keep it even after merge in dev, even though it won't be used (debatable).